### PR TITLE
chore: add examples to JsDoc for uint256.ts file

### PR DESCRIPTION
**Pull request by NueloSE**
## Motivation and Resolution

Provide developers with example of use case of each function

## Usage related changes
Running example code should produce same output as shown in example

## Development related changes

Documentation update

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

File: src/utils/uint256.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ export { UINT_128_MAX, UINT_256_MAX };
...
+ * @param uint256 Uint256 value to convert to bigint
+ * @returns {bigint} BigInt representation of the input Uint25
+ * @example
+ * ```typescript
+ * const uint256Value: Uint256 = "0x2fd23d9182193775423497fc0c472e156c57c69e4089a1967fb288a2d84e914";
+ * const result = uint256ToBN(uint256Value);
+ * // result = 28139218908760698114965631183142736658067655190345294133754686837148899816404n
+ * ```
  */
 export function uint256ToBN(uint256: Uint256) {
```
Comment (tabaktoni) [inline src/utils/uint256.ts:22]: Hm, Sir how is "0x2fd23d9182193775423497fc0c472e156c57c69e4089a1967fb288a2d84e914" Uint256,
Grr i need to inspect now in details all this PR's

File: src/utils/uint256.ts [hunk 2]
```diff
@@ -18,6 +26,14 @@ export function uint256ToBN(uint256: Uint256) {
 /**
  * Test BigNumberish is smaller or equal 2**256-1
```
Comment (tabaktoni) [inline src/utils/uint256.ts:27]: This claim is not accurate, the same for the claim below
Claim is (-infinite <= 2^256-1)
**accurate claim is:**
Test BigNumberish is in the range [0, 2^256-1]
https://stackoverflow.com/questions/4396290/what-does-this-square-bracket-and-parenthesis-bracket-notation-mean-first1-last

File: src/utils/uint256.ts [hunk 3]
```diff
@@ -26,6 +42,14 @@ export function isUint256(bn: BigNumberish): boolean {
 /**
  * Convert BigNumberish (string | number | bigint) to Uint256 (hex)
  * Legacy support Export
+ * @param bn BigNumberish value to convert to Uint256 (hex)
+ * @returns {Uint256} Uint256 (hex) representation of the input BigNumberish value
+ * @example
+ * ```typescript
+ * const bnValue: BigNumberish = 28139218908760698114965631183142736658067655190345294133754686837148899816404n;
+ * const result = bnToUint256(bnValue);
+ * // result = "0x2fd23d9182193775423497fc0c472e156c57c69e4089a1967fb288a2d84e914"
```
Comment (tabaktoni) [inline src/utils/uint256.ts:51]: This claim is false and can't be falser 🗡️

**Review by tabaktoni — CHANGES_REQUESTED** [review]
Not Good

File: src/utils/uint256.ts (hunk 3 above)
Comment (PhilippeR26) [reply src/utils/uint256.ts:51]: You have to take also into account that it has to be a code that a user can write and use.
bnToUint256() is not directly accessible for Starknet.js users.
To have to use : `uint256.bnToUint256()`

Comment (NueloSE) [reply src/utils/uint256.ts]: Appologies for the false calm. Corrections have been implemented

File: src/utils/uint256.ts (hunk 2 above)
Comment (NueloSE) [reply src/utils/uint256.ts]: Thank you for pointing this out. This description came with the old version.
I have implemented the correct thing now.

File: src/utils/uint256.ts (hunk 1 above)
Comment (NueloSE) [reply src/utils/uint256.ts:22]: Applogies for this error. The Hex-decimal string has now be converted a `CairoUint256` object. Such mistake won't occur again going forward.

File: src/utils/uint256.ts [hunk 4]
```diff
@@ -10,14 +10,34 @@ export { UINT_128_MAX, UINT_256_MAX };
 /**
  * Convert Uint256 to bigint
  * Legacy support Export
+ * @param uint256 Uint256 value to convert to bigint
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:13]: Uint256 has to be the first word after @param, and has to be wrapped with {}.

File: src/utils/uint256.ts [hunk 5]
```diff
@@ -10,14 +10,34 @@ export { UINT_128_MAX, UINT_256_MAX };
...
+ * @returns {bigint} BigInt representation of the input Uint25
+ * @example
+ * ```typescript
+ * const uint256Value: Uint256 = { low: 1234567890, high: 1 };
+ * const result = uint256ToBN(uint256Value);
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:18]: All these functions are in the namespace `uint256`. 
Users have to be able to perform these tests.
Here, without the namespace, it's not possible.
I don't understand how you have succeeded to test these examples!

File: src/utils/uint256.ts [hunk 6]
```diff
@@ -10,14 +10,34 @@ export { UINT_128_MAX, UINT_256_MAX };
 /**
  * Convert Uint256 to bigint
  * Legacy support Export
+ * @param uint256 Uint256 value to convert to bigint
+ * @returns {bigint} BigInt representation of the input Uint25
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:14]: `6` missing at the end.

File: src/utils/uint256.ts [hunk 7]
```diff
@@ -10,14 +10,34 @@ export { UINT_128_MAX, UINT_256_MAX };
...
+ * @example
+ * ```typescript
+ * const uint256Value: Uint256 = { low: 1234567890, high: 1 };
+ * const result = uint256ToBN(uint256Value);
+ * // result = 519229685234822890n
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:19]: I have executed your example, and I don't have the same result : 
`340282366920938463463374607433002779346n`
You have to be more attentive ...

File: src/utils/uint256.ts [hunk 8]
```diff
@@ -10,14 +10,34 @@ export { UINT_128_MAX, UINT_256_MAX };
...
  */
 export function uint256ToBN(uint256: Uint256) {
   return new CairoUint256(uint256).toBigInt();
 }
 
 /**
- * Test BigNumberish is smaller or equal 2**256-1
+ * Test BigNumberish is in the range[0, 2**256-1]
  * Legacy support Export
+ * @param bn BigNumberish value to test
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:29]: bn wrapped with {]

File: src/utils/uint256.ts [hunk 9]
```diff
@@ -26,6 +46,20 @@ export function isUint256(bn: BigNumberish): boolean {
 /**
  * Convert BigNumberish (string | number | bigint) to Uint256 (hex)
  * Legacy support Export
+ * @param bn BigNumberish value to convert to Uint256 (hex)
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:49]: bn under {}

File: src/utils/uint256.ts [hunk 10]
```diff
@@ -26,6 +46,20 @@ export function isUint256(bn: BigNumberish): boolean {
...
+ * @returns {Uint256} Uint256 (hex) representation of the input BigNumberish value
+ * @example
+ * ```typescript
+ * import { uint256 } from "path/to/file";
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:53]: This time we have import of the namespace. Here it's correct...
except that its `from 'starknet'`.

File: src/utils/uint256.ts [hunk 11]
```diff
@@ -26,6 +46,20 @@ export function isUint256(bn: BigNumberish): boolean {
...
+ * 
+ * const bnValue: BigNumberish = -1n;
+ * const result = uint256.bnToUint256(bnValue);
+ * // Throws Error: bigNumberish is smaller than UINT_256_MIN
+ * 
+ * const bnValue: BigNumberish = 1000000000n;
+ * const result1 = uint256.bnToUint256(bnValue);
+ * // result1 = { "high": "0x0", "low": "0x3b9aca00" }
```
Comment (PhilippeR26) [inline src/utils/uint256.ts:61]: A Uint256 has always the `low` property at the beginning.

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
To correct.

**Pull request closed without merging**